plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,974 @@ type CiscoTelemetryMDT struct {
 
 	// Internal state
 	aliases   map[string]string
+	dmes      map[string]string
 	warned    map[string]struct{}
 	extraTags map[string]map[string]struct{}
+	nxpathMap map[string]map[string]string //per path map
+	propMap   map[string]func(field *telemetry.TelemetryField, value interface{}) interface{}
 	mutex     sync.Mutex
 	acc       telegraf.Accumulator
 	wg        sync.WaitGroup
 }
 
+type NxPayloadXfromStructure struct {
+	Name string `json:"Name"`
+	Prop []struct {
+		Key   string `json:"Key"`
+		Value string `json:"Value"`
+	} `json:"prop"`
+}
+
+//xform Field to string
+func xformValueString(field *telemetry.TelemetryField) string {
+	var str string
+	switch val := field.ValueByType.(type) {
+	case *telemetry.TelemetryField_StringValue:
+		if len(val.StringValue) > 0 {
+			return val.StringValue
+		}
+	case *telemetry.TelemetryField_Uint32Value:
+		str = strconv.FormatUint(uint64(val.Uint32Value), 10)
+		return str
+	case *telemetry.TelemetryField_Uint64Value:
+		str = strconv.FormatUint(val.Uint64Value, 10)
+		return str
+	case *telemetry.TelemetryField_Sint32Value:
+		str = strconv.FormatInt(int64(val.Sint32Value), 10)
+		return str
+	case *telemetry.TelemetryField_Sint64Value:
+		str = strconv.FormatInt(val.Sint64Value, 10)
+		return str
+	}
+	return ""
+}
+
+//xform Uint64 to int64
+func (c *CiscoTelemetryMDT) nxosValueXformUint64Toint64(field *telemetry.TelemetryField, value interface{}) interface{} {
```

Comment:
I think you don't use `c` so this doesn't have to be a member function of `CiscoTelemetryMDT`. You might still want to keep as is but then please also change `xformValueString` to follow the same pattern.